admin/collection_providers/views.py

```diff
@@ -233,36 +249,19 @@ class CollectionProviderChangeForm(PermissionRequiredMixin, UpdateView):
     model = CollectionProvider
     form_class = CollectionProviderForm
 
-    def form_valid(self, form):
-        if self.object.primary_collection:
-            self.object.primary_collection.collected_type_choices.extend(form.cleaned_data['collected_type_choices']['added'])
-            for item in form.cleaned_data['collected_type_choices']['removed']:
-                self.object.primary_collection.collected_type_choices.remove(item)
+    def process_choices(self, choices_name, form):
```

Comment:
You're passing enough stuff here already that you might as well just pass the collection and make this a module-level function instead of keeping it in the class?